src/Uno.UI.Runtime.Skia.Gtk/GtkCoreWindowExtension.cs

```diff
@@ -84,6 +87,62 @@ public GtkCoreWindowExtension(object owner)
 
 		partial void InitializeKeyboard();
 
+		internal static Fixed? FindNativeOverlayLayer(Gtk.Window window)
+		{
+			var overlay = (Overlay)((EventBox)window.Child).Child;
+			return overlay.Children.OfType<Fixed>().FirstOrDefault();
+		}
+
+		internal static Fixed? GetOverlayLayer(XamlRoot xamlRoot) =>
+			XamlRootMap.GetHostForRoot(xamlRoot)?.NativeOverlayLayer;
+
+		public bool IsNativeElement(object content)
+			=> content is Widget;
+
+		public void AttachNativeElement(object owner, object content)
+		{
+			if (content is Widget widget
+				&& owner is XamlRoot xamlRoot
+				&& GetOverlayLayer(xamlRoot) is { } overlay)
+			{
+				widget.ShowAll();
+				overlay.Put(widget, 0, 0);
+			}
```

Comment:
Should we log something if not of the expected type (especially is not a `Widget` as it's also referenced as an `object` in the `ContentPresenter`)?